pkgs/development/node-packages/default.nix

```diff
@@ -119,6 +119,7 @@ let
     # NOTE: this is a stub package to fetch npm dependencies for
     # ../../applications/video/epgstation
     epgstation = super."epgstation-../../applications/video/epgstation".override (drv: {
+      nativeBuildInputs = [ self.node-pre-gyp self.node-gyp-build ];
```

Comment:
Shouldn't this be a `buildInputs` instead?

Reply:
`node-pre-gyp` and `node-gyp-build` are used at build time for compiling native binaries, so I thought it belonged to `nativeBuildInputs`.

Reply:
Yeah, this is exactly why I think they're `buildInputs`. The `native` from `nativeBuildInputs` is not related to "native" compilation, but the fact that it will run with the same architecture as the build platform (e.g.: the platform building the code), that may be different from the host platform (e.g.: the platform that will actually run the code).

Now this difference only matters for cross-compilation AFAIK, but when cross compiling you can build in, say, `aarch64-linux` code from a `x86_64-linux` machine. The way it currently is, it would use `node-pre-gyp` and `node-gyp-build` from `x86_64-linux` to build `aarch64-linux` code and it seems wrong to me.

`nativeBuildInputs` are generally for things that are related to code generation (where the architecture will not matter because the result should be the same), for example `make`.

Reply:
>  The way it currently is, it would use `node-pre-gyp` and `node-gyp-build` from x86_64-linux to build aarch64-linux code 

Isn't this correct behavior? `node-pre-gyp` and `node-gyp-build` runs at build time, so it has to be targeted for the build platform, not the runtime platform.

Reply:
From what I understand, no, because the only thing here that should run in the build platform during cross-compilation is the compiler and pre-processors. The dependencies itself should be build to the target platform, and this include `node-pre-gyp` and `node-gyp-build`.

That is, unless `node-pre-gyp` and `node-gyp-build` are just pre-processors or something that will generate the code to be build. Them yes, they should be `nativeBuildInputs`.

Reply:
I am going to say that I am not 100% sure right now, but looking at the other derivations in the same file, most (all?) of them put `node-pre-gyp` and `node-gyp-build` to `buildInputs`. So I think it is safer to leave it there.

Reply:
For more context, both are build-time only dependencies and never run on the target platform. But it may not be relevant because having looked at it more closely, I doubt node packages can be cross compiled in its current state. For cross compilation to work, node2nix would need to use 2 versions of node: one for the build platform and the other for the target platform.

Anyways, I've now replaced `nativeBuildInputs` with `buildInputs`.